### ISSUE TITLE
container_name option

### ISSUE THREAD

**Issue by mcctomsk** [docker/docker.github.io#2950]
Hello.
File: [compose/compose-file/index.md](https://docs.docker.com/compose/compose-file/), CC @londoncalling 

$ docker-compose --version
```
docker-compose version 1.11.2, build dfed245

```
In documentation you wrote:
> container_name: my-web-container
> Because Docker container names must be unique, you cannot scale a service beyond 1 container if you have specified a custom name. Attempting to do so results in an error.

But when I've tried to use container_name option I've got next warning: 
```
$ docker stack deploy -c docker-compose.yaml --with-registry-auth=true green
Ignoring deprecated options:

container_name: Setting the container name is not supported.
```
Probably there need to add your standard note too:

> This option is ignored when deploying a stack in swarm mode with a (version 3) Compose file.

**Comment by joaofnfernandes** [comment]
Yup. That doc should say that `container_name` is not supported if you're deploying with `docker stack deploy`. We should also validate that this doc is correct for other [keywords that are not supported](https://github.com/moby/moby/blob/master/cli/compose/types/types.go#L34).

**Comment by londoncalling** [comment]
@mcctomsk @joaofnfernandes thanks. From what I can tell, this `container_name` would inherently not make sense with `docker stack deploy` _or_ by v3 at all, so I've worded the note a little differently.

I've submitted a PR ^^ and waiting for some clarification/verify from the Compose developers.

**Comment by londoncalling** [comment]
@mcctomsk thank, fixed by PR #3003, should be published soon.
